_use_cases/encryption-at-rest.markdown

```diff
@@ -0,0 +1,105 @@
+
+This present a challenge for an enterprise using Kafka to distribute confidential data or PII (Personally Identifiable
+Information).  Enterprises are often subject to confidential requirements coming from governmental bodies such as
+GDPR in the EU or LGPD in Brazil, industry standards such as HIPAA in the health domain or PCI/DSS in the payments domain,
+in addition to any in-house security and compliance requirements.
+
+The problem is made more complex if the enterprise has opted to utilise a Cloud Kafka Service as the confidential
+data is now residing in the clear on the file systems of the service provider.
+
+| ![image](../assets/encryption-at-rest_problem.png){:width="100%"} |
+|:-----------------------------------------------------------------:|
+|    *Problem: Plain text records readable by the Kafka Admins*     |
+
+### Isn't TLS sufficient?
+
+TLS encrypts the content _on the wire_.  It means that someone using a network sniffer cannot tap what is being
+sent over the wire between the application and the Kafka Broker.  However, once the network packets arrive at the broker,
+the packets are decrypted and exist in the clear once again.  This means the confidential records are in the clear in the memory
+of the broker and in the clear when the data is written to the file system.
+
+TLS does not change the problem.
+
+### Isn't storage volume encryption an answer?
+
+With storage volume encryption, the contents of the device are encrypted with a key.  This approach provides some mitigations.
+If the storage device is stolen or the storage device hijacked and attached to an attacker's computer, the attacker won't have
+the encryption key so won't be able to read the data, including the Kafka records.
+
+However, there are shortcomings.  The encrypted storage volume must be mounted to the computers running the Kafka broker. To do
+this the  and the encryption keys applied at the operating system level.  Anybody with shell level access to the hosts  is likely
+to be able to read the data, including the Kafka confidential records.
+
+Storage volume encryption doesn't really solve the problem.
+
+### Can't the applications encrypt/decrypt the data?
+
+It is possible for producing applications to encrypt data before sending it to Kafka, and for consuming applications to decrypt it
+again.  With this approach the brokers never possess the records in the clear and as they don't have encryption keys, they cannot
+decrypt it.
+
+So, this approach does offer a solution to the problem but there are disadvantages.
+
+1. Kafka Client libraries don't support encryption themselves.
+1. Kafka Client ecosystem is polyglot - it is common to Kafka applications written in many languages (Java, Go, Rust...)
+   deployed within a _single organisation_.  Any solution needs to work across all the target languages.
+1. There are some encryption libraries for some languages that can be used with the Kafka Clients but these available
+   across all languages.
```

Comment:
I'm not sure if this a separate bullet point or part of the following paragraph. 

The compliance costs of implementing encryption and decryption `n` times, security audits etc.

There is a lot to be said for `do it once and do it right`.